src/Microsoft.AspNet.Mvc.Razor/Precompilation/CodeAnalysisAttributeUtilities.cs

```diff
@@ -0,0 +1,256 @@
+
+            for (var index = 0; index < matchInfo.ArgumentValues.Length; index++)
+            {
+                parameters[index] =
+                    Expression.Convert(
+                        Expression.ArrayIndex(
+                            argsParameter,
+                            Expression.Constant(index)),
+                        matchInfo.ArgumentValues[index].GetType());
+            }
+
+
+            // () => new TAttribute(args)
+            var lambda =
+                Expression.Lambda<Func<object[], Attribute>>(
+                    Expression.New(
+                        matchInfo.Constructor,
+                        parameters),
+                    argsParameter);
+
+            return lambda.Compile();
+        }
+
+        private static ConstructorMatchInfo MatchConstructor(
+            Type type,
+            AttributeData attributeData,
+            CodeAnalysisSymbolLookupCache symbolLookup)
+        {
+            var constructor = FindConstructor(type, attributeData.ConstructorArguments, symbolLookup);
+            var constructorParmaters = constructor.GetParameters();
+
+            var arguments = new object[attributeData.ConstructorArguments.Length];
+            for (var i = 0; i < arguments.Length; i++)
+            {
+                var value = ConvertTypedConstantValue(
+                    constructorParmaters[i].ParameterType,
+                    attributeData.ConstructorArguments[i],
+                    symbolLookup);
+
+                arguments[i] = value;
+            }
+
+            return new ConstructorMatchInfo
+            {
+                Constructor = constructor,
+                ArgumentValues = arguments
+            };
+        }
+
+        private static ConstructorInfo FindConstructor(
+            Type type,
+            ImmutableArray<TypedConstant> symbolConstructorArguments,
+            CodeAnalysisSymbolLookupCache symbolLookup)
+        {
+            var constructors = type.GetConstructors();
+            // One or more constructor arguments were specified. Consequently,
+            // we must have at least one constructor that matches.
+            Debug.Assert(constructors.Length != 0);
+
+            if (constructors.Length == 1)
+            {
+                return constructors[0];
+            }
+
+            foreach (var constructor in constructors)
+            {
+                var runtimeParameters = constructor.GetParameters();
+                if (runtimeParameters.Length != symbolConstructorArguments.Length)
+                {
+                    continue;
+                }
+
+                var parametersMatched = true;
+                for (var index = 0; index < runtimeParameters.Length; index++)
+                {
+                    var runtimeParameter = runtimeParameters[index].ParameterType;
+                    if (symbolConstructorArguments[index].Kind == TypedConstantKind.Array &&
```

Comment:
You're comparing `index` for `index` here. Will this handle named arguments appropriately? Aka

```C#
[SomeAttribute(secondAttribute: 2, firstAttribute: 1)]
```